File: BPGezinswetenschappen.Core/Models/Organisation.cs
namespace BPGezinswetenschappen.DAL.Models
{
    public class Organisation
    {
        public int OrganisationId { get; set; }
        public string Name { get; set; }
        public string? Address { get; set; }
        public string? PostalCode { get; set; }
        public string? City { get; set; }
        public string? Phone { get; set; }
        public string? Email { get; set; }
        public string? Url { get; set; }
        public string? Contact { get; set; }

        // navigation property

        public ICollection<Project>? Projects { get; set; }


    }
}

File: Client/src/app/shared/models/organisation.ts
import { Project } from './project';

export interface Organisation {
  organisationId: number;
  name: string;
  address?: string;
  postalCode?: string;
  city?: string;
  phone?: string;
  email?: string;
  url?: string;
  contact?: string;
  projects?: Project[];
}

export interface OrganisationCreateDto {
  name: string;
  address?: string;
  postalCode?: string;
  city?: string;
  phone?: string;
  email?: string;
  url?: string;
  contact?: string;
}

export interface OrganisationReadDto {
  organisationId: number;
  name: string;
  address: string;
  postalCode: string;
  city: string;
  phone: string;
  email: string;
  url: string;
  contact: string;
}

export interface OrganisationUpdateDto {
  name?: string;
  address?: string;
  postalCode?: string;
  city?: string;
  phone?: string;
  email?: string;
  url?: string;
  contact?: string;
}
